src/Core/Service/MessengerRetryStrategy.php

```diff
@@ -0,0 +1,114 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Service;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Update\Event\UpdatePostFinishEvent;
+use Shopware\Core\Framework\Webhook\Message\WebhookEventMessage;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Retry\RetryStrategyInterface;
+use Symfony\Component\Messenger\Stamp\RedeliveryStamp;
+use Symfony\Contracts\Service\ResetInterface;
+
+/**
+ * @internal
+ */
+#[Package('core')]
+class MessengerRetryStrategy implements RetryStrategyInterface, ResetInterface
```

Comment:
did you though about making a PR to symfony so we have a event 🤔  (this is still fine, but maybe we can get rid of it with Symfony 7.3 then)

Reply:
There is already this event `WorkerMessageFailedEvent` but I could not figure out or see something in the docs where I can set the delay on it.

Reply:
I will dig a little more and maybe open discussion at symfony

Reply:
@shyim what do you think about instead: I could dispatch our own event in `\Shopware\Core\Framework\Webhook\Handler\WebhookEventMessageHandler` when we have a failure, there I can add listener in service bundle and throw a `\Symfony\Component\Messenger\Exception\RecoverableMessageHandlingException` with a delay and symfony would pick it up and auto retry. 

I think that's better, can probably get rid of compiler pass and strategy

Reply:
done, but need to wait on a bugfix at Symfony